dali/c_api/c_api.cc

```diff
@@ -105,20 +105,32 @@ void daliCreatePipeline(daliPipelineHandle* pipe_handle,
     pipe->SetQueueSizes(cpu_prefetch_queue_depth, gpu_prefetch_queue_depth);
   }
   pipe->Build();
-  pipe_handle->pipe = reinterpret_cast<void*>(pipe);
+  pipe_handle->pipe = reinterpret_cast<void *>(pipe);
   pipe_handle->ws = new dali::DeviceWorkspace();
   CUDA_CALL(cudaStreamCreateWithFlags(&pipe_handle->copy_stream, cudaStreamNonBlocking));
 }
 
-void daliPrefetchUniform(daliPipelineHandle* pipe_handle, int queue_depth) {
-  dali::Pipeline* pipeline = reinterpret_cast<dali::Pipeline*>(pipe_handle->pipe);
+
+void daliDeserializeDefault(daliPipelineHandle *pipe_handle, const char *serialized_pipeline,
+                            int length) {
+  dali::Pipeline *pipe = new dali::Pipeline(std::string(serialized_pipeline, length));
```

Comment:
do we allow length < 0 to denote a null-terminated string? If so, you'd need to include that in string construction:
```suggestion
  std::string str = length >= 0 ? std::string(serialized_pipeline, length) : std::string(serialized_pipeline)
  dali::Pipeline *pipe = new dali::Pipeline(std::move(str));
```

Reply:
We decided not to do this, since the pipeline can contain `\0`'s inside